purchase_package_qty/models/stock_picking.py

```diff
@@ -95,13 +83,16 @@ def prepare_vals_order_line(self, diff_pack_op):
                 'product_qty_package': diff_pack_op.product_qty_package,
                 'package_qty': diff_pack_op.package_qty,
                 'order_id': order.id,
-                'product_uom': diff_pack_op.product_uom_id.id,
+                'product_uom': diff_pack_op.product_uom.id,
                 'price_unit': price_unit,
                 'date_planned': order.date_planned,
                 'taxes_id': [(
                         6, 0, [x.id for x in
                                diff_pack_op.product_id.supplier_taxes_id])],
-                'operation_extra_id': diff_pack_op.id
             })
-            po_line = po_line_env.create(vals)
+            po_line = po_line_env.with_context(
+                skip_move_create=True).create(vals)
+            order.message_post(body=_('purchase items : %s with %s qty. \
+                were created from incoming shipment (%s).') % (
+                name, diff_pack_op.package_qty, self.origin,))
```

Comment:
@chetanrdhaduk as a general rule, try to avoid as much as you can using '\'.
You can do it like this, for instance
```suggestion
            order.message_post(body=_(
                'Purchase items: %s with %s qty. were created from '
                'incoming shipment (%s).') % (
                    name, diff_pack_op.package_qty, self.origin))
```